pkg/network/usm/sharedlibraries/testutil/file_opener.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testutil
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// mutex protecting build process
+var mux sync.Mutex
+var cachedArtifactPath string
+
+func OpenFromAnotherProcess(t *testing.T, paths ...string) *exec.Cmd {
+	build(t)
```

Comment:
the code cannot work in the CI, as we cannot build in the context of kitchen tests (missing go binary, etc.)
that's why in the original code we had a setup for the ci + building the sowatcher client in during `system-probe.kitchen-prepare`

Reply:
I refactored the code and documented this, since it wasn't really clear in the code